Simulator/Test/test_sun_sensor_math.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+"""
+Created on Sat Nov 30 17:22:15 2019
+
+"""
+import pytest
+import os, sys, inspect
+import sun_sensor_math as LA
+import numpy as np
+
+currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+groundtruth_dir = os.path.dirname(currentdir)
+sys.path.insert(0, groundtruth_dir)
+
+def pytest_addoption(parser):
+    parser.addoption(
+        "--cmdopt", action="store", default="type1", help="my option: type1 or type2"
+    )
+
+@pytest.fixture
+def cmdopt(request):
+    return request.config.getoption("--cmdopt")
+
+def test_sense2vector_error():
+    with pytest.raises(TypeError):
+        measurements = 5
+        r_Earth2Sun = [1,1,1]
+        r_sat = [1,1,1]
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+        
+        measurements = np.array([5,5,5])
+        r_Earth2Sun = [1,1,1]
+        r_sat = [1,1,1]
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+        
+        measurements = [5,3,3,4,5,6]
+        r_Earth2Sun = 1
+        r_sat = [1,1,1]
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+        
+        measurements = [5,3,3,4,5,6]
+        r_Earth2Sun = np.array([1,1,1])
+        r_sat = [1,1,1]
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+        
+        measurements = [5,3,3,4,5,6]
+        r_Earth2Sun = [1,1,1]
+        r_sat = 1
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+        
+        measurements = [5,3,3,4,5,6]
+        r_Earth2Sun = [1,1,1]
+        r_sat = np.array([1,1,1])
+        assert LA.sense2vector(measurements, r_Earth2Sun, r_sat)
+    
+def test_vector2sense():
+    assert True
+    
+def test_deltas2measure():
+    deltas = [1,-1,1]
+    assert LA.deltas2measure(deltas) == [1,0,0,1,1,0]
+    
+    deltas = [-1,-1,1]
+    assert LA.deltas2measure(deltas) == [0,1,0,1,1,0]
+    
+    deltas = [-100,1,1]
+    assert LA.deltas2measure(deltas) == [0,100,1,0,1,0]
+    
+    deltas = [-100,1]
+    assert LA.deltas2measure(deltas) == [0,100,1,0]
+    
+    deltas = [-100]
+    assert LA.deltas2measure(deltas) == [0,100]
+    
+    
+def test_isEclipse2():
+    measurements = [3,3,3,3,3,3]
+    thresh = 5
+    assert LA.isEclipse2(measurements, thresh) == True
```

Comment:
I just took a look at `isEclipse2` and it looks like a redundant function. It does the same thing as the built-in function `all`.

Reply:
See:
https://github.com/spacecraft-design-lab-2019/groundtruth-simulator/blob/2b6c4eeb89762eaab40abced4b1675b3fcceb9f3/Simulator/sun_sensor_math.py#L138-L153

Reply:
What are you suggesting? That inside of isEclipse2 we use all(), or we never use isEclipse2 and we just use all() wherever we need to check for whether the satellite is in eclipse?

Reply:
The second option. There is no "substance" to isEclipse2. It's just an elementwise greater-than check. It also isn't used yet anywhere, so this would have no impact.